userspace/libsinsp/filter.cpp

```diff
@@ -1038,6 +1051,22 @@ char* sinsp_filter_check::rawval_to_string(uint8_t* rawval,
 					 sizeof(m_getpropertystr_storage),
 					 "%.1lf", *(double*)rawval);
 			return m_getpropertystr_storage;
+		case PT_CHARBUFARRAY:
+			getpropertystr = "(";
+			strvect = (vector<char*>*)*(uint64_t *)rawval;
+			for (auto it = strvect->begin(); it != strvect->end(); ++it)
+			{
+				if (it != strvect->begin())
+				{
+					getpropertystr += ",";
+				}
+				getpropertystr += *it;
+			}
+			getpropertystr += ")";
+			strncpy(m_getpropertystr_storage,
```

Comment:
I think you could make this simpler by adding a generic m_getproperty_strval or similar to sinsp_filter_check, building a string containing the values as you're doing, and then returning a `.c_str()` cast to non-const.

A lot of other filterchecks do a similar thing like sinsp_filter_check_user and m_strval, sinsp_filter_check_tracer and m_strstorage, etc.